### PR TITLE
fix: [CDS-84742]: Remove redundant annotation from step/stage nodes v1

### PR DESCRIPTION
**Pull request by shalini-agr**
## Describe your changes
**Changes:** Removing the unwanted annotations from v1 POJOs. 

**Testing:** Ran a pipeline locally

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55615)

### CODE REVIEW

**Comment by shalini-agr** [comment]
trigger smartchecks

**Review by archit-harness — DISMISSED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55615)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Comment by shalini-agr** [comment]
trigger uts

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55615)* status: :shipit: complete! ✓

**Comment by shalini-agr** [comment]
trigger uts

**Comment by shalini-agr** [comment]
trigger uts

**Review by archit-harness — APPROVED** [review]
> Reviewed 3 of 3 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55615)* status: 6 file(s) to review. 1 of 2 approvals obtained.

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/creator/plan/stage/v1/DeploymentStageNodeV1.java [hunk 1]
```diff
@@ -16,12 +15,9 @@
 import com.fasterxml.jackson.annotation.JsonTypeName;
 import javax.validation.constraints.NotNull;
 import lombok.Value;
-import org.springframework.data.annotation.TypeAlias;
 
 @Value
 @JsonTypeName(YAMLFieldNameConstants.DEPLOYMENT_STAGE_V1)
-@TypeAlias("DeploymentStageNodeV1")
-@RecasterAlias("io.harness.cdng.creator.plan.stage.v1.DeploymentStageNodeV1")
 @CodePulse(module = ProductModule.CDS, unitCoverageRequired = false, components = {HarnessModuleComponent.CDS_PIPELINE})
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/creator/plan/stage/v1/DeploymentStageNodeV1.java:21]: why are we removing the recaster alias ?

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55615)* status: 6 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/creator/plan/stage/v1/DeploymentStageNodeV1.java` line 21 at r3](https://reviewable.io/reviews/harness/harness-core/55615#gh-1398813142:-NjfsS8_8L0Ijc4rr6TB:b-sf90o5) ([raw file](https://github.com/harness/harness-core/blob/46b45d3c4bd1396f6f2f31048a39e5b5ea12f699/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/creator/plan/stage/v1/DeploymentStageNodeV1.java#L21)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

why are we removing the recaster alias ?
</blockquote></details>

we need the recaster alias annotation to ensure backward compatibility for deserialization in case of movement of classes but this class is just a POJO which is used to deserialize yaml during plan creation and read fields from it since it is all sequential and won't be affected by movement of classes, so, it's not needed here. 
We just need it on the step parameters and all classes used in it.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 5 of 5 files at r1, 3 of 3 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55615)* status: :shipit: complete! ✓

**Review by raghavharness — APPROVED** [review]

**Comment by shalini-agr** [comment]
trigger review

**Pull request merged**